src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticConverter.cs

```diff
@@ -24,7 +25,18 @@ public static Diagnostic Convert(RazorDiagnostic razorDiagnostic, SourceText sou
             throw new ArgumentNullException(nameof(sourceText));
         }
 
-        var diagnostic = new Diagnostic()
+        var projects = documentSnapshot is null
```

Comment:
If this is `projects`, plural, should we go back to the project snapshot manager and find all of them? Or maybe snapshot manager can do that? We do something similar in completion like that, to show which tag helpers are available.

Reply:
I had plural only because it is an array in the VSDiagnostic. Do we currently report different diagnostics for different TFM? I didn't think so yet. Likely that will be needed

Reply:
Different diagnostics for different TFMs are probably possible right now, but still unlikely and , I've got no idea how/if we handle them. Probably fine for a follow up.

If the platform requests diagnostics for us multiple times per document, with a different project context on each, then we're probably fine. If it doesn't do that, then we probably need to ignore the project context on the request and pull all diagnostics for the document from every project and consolidate. No idea which is correct though :)